File: src/models/Poll.ts
import mongoose, { Document, Schema } from "mongoose";

export interface IPoll extends Document {
  messageId: string;
  channelId: string;
  guildId: string;
  creatorId: string;
  question: string;
  options: Array<{
    letter: string;
    emoji: string;
    text: string;
  }>;
  anonymous: boolean;
  createdAt: Date;
  endsAt: Date | null;
  votes: Record<string, string>; // userId -> selectedOption
}

const PollSchema = new Schema({
  messageId: { type: String, required: true, unique: true },
  channelId: { type: String, required: true },
  guildId: { type: String, required: true },
  creatorId: { type: String, required: true },
  question: { type: String, required: true },
  options: [
    {
      letter: String,
      emoji: String,
      text: String,
    },
  ],
  anonymous: { type: Boolean, default: false },
  createdAt: { type: Date, default: Date.now },
  endsAt: { type: Date, default: null },
  votes: { type: Schema.Types.Mixed, default: {} },
});

export default mongoose.model<IPoll>("Poll", PollSchema);

File: Models/Poll.cs
namespace KaorukoBot.Models;

public class Poll
{
    public string MessageId { get; set; } = string.Empty;
    public string ChannelId { get; set; } = string.Empty;
    public string GuildId { get; set; } = string.Empty;
    public string Question { get; set; } = string.Empty;
    public List<string> Options { get; set; } = [];
    public Dictionary<string, int> Votes { get; set; } = []; // userId -> optionIndex
    public bool Anonymous { get; set; }
    public int DurationMinutes { get; set; } = 60;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string CreatedBy { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
}
